spec/single_city_spec.rb

```diff
@@ -0,0 +1,98 @@
+ require 'spec_helper'
+
+ describe 'Open Weather API ' do
+ 
+  before(:all) do
+  @json_data = Generator.new.city
+  @random_city_info = Generator.new.city.generate_random_cities(rand(0..(@json_data.array_length)))
```

Comment:
I would have had a method in your generator where you could have added this logic so you could have had a simple method to get a random city id. You shouldn't have to worry about doing the logic for this in the spec file. The whole idea of generator is it will abstract this logic out for us.